### PR TITLE
refactor(doctor): migrate to CA basecode

### PR DESCRIPTION
**Pull request by thisham**
## Overview
This migration will be affect to the old architecture (MVC). Models are moved to repositories. Then, related dependencies of MVC's models are referenced to feature stage repositories. 

Therefore this PR can't directly reviewed nor merged. So that, prefer to merge or wait until this PR #27 approved first.

## Testing
- [x] Get all doctors
- [x] Get doctor by ID
- [x] Add new doctor
- [x] Amend existing doctor by ID
- [x] Remove existing doctor by ID

### CODE REVIEW

File: src/app/doctor/services/usecases_test.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+	t.Run("should got data by id", func(t *testing.T) {
+		mockRepo.On("SelectDataByID", sampleUUIDIkoUwais.String()).Return(&sampleDomainIkoUwais, nil).Once()
+		result, err := services.GetDoctorByID(sampleUUIDIkoUwais.String())
+
+		assert.Nil(t, err)
+		assert.Equal(t, sampleDomainIkoUwais.Name, result.Name)
+	})
+
+	t.Run("should got database error", func(t *testing.T) {
+		mockRepo.On("SelectDataByID", sampleUUIDIkoUwais.String()).Return(nil, errors.New("can't connect to the database")).Once()
```
Comment (dhichii) [inline src/app/doctor/services/usecases_test.go:131]: prefer to add Return in a new line, so the reviewer doesn't need to swipe to see the whole code

File: src/app/doctor/services/usecases_test.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+	t.Run("should created a data", func(t *testing.T) {
+		mockRepo.On("InsertData", sampleDomainIkoUwaisWithNoUUID).Return(sampleUUIDIkoUwais.String(), nil).Once()
+		result, err := services.CreateDoctor(sampleDomainIkoUwaisWithNoUUID)
+
+		assert.Nil(t, err)
+		assert.Equal(t, sampleUUIDIkoUwais.String(), result)
+	})
+
+	t.Run("should got database error", func(t *testing.T) {
+		mockRepo.On("InsertData", sampleDomainIkoUwaisWithNoUUID).Return(uuid.Nil.String(), errors.New("can't connect to the database")).Once()
```
Comment (dhichii) [inline src/app/doctor/services/usecases_test.go:157]: prefer to add Return in a new line, so the reviewer doesn't need to swipe to see the whole code

File: src/app/doctor/services/usecases_test.go [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+func TestUpdateDoctorByID(t *testing.T) {
+	t.Run("should update data by id", func(t *testing.T) {
+		mockRepo.On("UpdateByID", sampleUUIDIkoUwais.String(), sampleDomainIkoUwaisWithNoUUID).Return(nil).Once()
+		err := services.AmendDoctorByID(sampleUUIDIkoUwais.String(), sampleDomainIkoUwaisWithNoUUID)
+
+		assert.Nil(t, err)
+	})
+
+	t.Run("should got database error", func(t *testing.T) {
+		mockRepo.On("UpdateByID", sampleUUIDIkoUwais.String(), sampleDomainIkoUwaisWithNoUUID).Return(errors.New("can't connect to the database")).Once()
```
Comment (dhichii) [inline src/app/doctor/services/usecases_test.go:174]: prefer to add Return in a new line, so the reviewer doesn't need to swipe to see the whole code

File: src/app/doctor/services/usecases_test.go [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+		err := services.AmendDoctorByID(sampleUUIDIkoUwais.String(), sampleDomainIkoUwaisWithNoUUID)
+
+		assert.NotNil(t, err)
+	})
+
+	t.Run("should got not found error", func(t *testing.T) {
+		mockRepo.On("UpdateByID", sampleUUIDIkoUwais.String(), sampleDomainIkoUwaisWithNoUUID).Return(errors.New("record not found")).Once()
```
Comment (dhichii) [inline src/app/doctor/services/usecases_test.go:181]: prefer to add Return in a new line, so the reviewer doesn't need to swipe to see the whole code

File: src/app/doctor/services/usecases_test.go (hunk 1 above)
Comment (dhichii) [inline src/app/doctor/services/usecases_test.go:131]: prefer to add Return in a new line, so the reviewer doesn't need to swipe to see the whole code

File: src/app/doctor/services/usecases_test.go (hunk 2 above)
Comment (dhichii) [inline src/app/doctor/services/usecases_test.go:157]: prefer to add Return in a new line, so the reviewer doesn't need to swipe to see the whole code

File: src/app/doctor/services/usecases_test.go (hunk 3 above)
Comment (dhichii) [inline src/app/doctor/services/usecases_test.go:174]: prefer to add Return in a new line, so the reviewer doesn't need to swipe to see the whole code

File: src/app/doctor/services/usecases_test.go (hunk 4 above)
Comment (dhichii) [inline src/app/doctor/services/usecases_test.go:181]: prefer to add Return in a new line, so the reviewer doesn't need to swipe to see the whole code

**Review by dhichii — CHANGES_REQUESTED** [review]

File: src/app/doctor/services/usecases_test.go (hunk 1 above)
Comment (thisham) [reply src/app/doctor/services/usecases_test.go]: applied https://github.com/ALTERA-CAPSTON-41/backend/pull/28/commits/ab70ccc7d3a194960c59f0075e1f98168f4477ca.

File: src/app/doctor/services/usecases_test.go (hunk 2 above)
Comment (thisham) [reply src/app/doctor/services/usecases_test.go]: applied https://github.com/ALTERA-CAPSTON-41/backend/pull/28/commits/ab70ccc7d3a194960c59f0075e1f98168f4477ca.

File: src/app/doctor/services/usecases_test.go (hunk 3 above)
Comment (thisham) [reply src/app/doctor/services/usecases_test.go]: applied https://github.com/ALTERA-CAPSTON-41/backend/pull/28/commits/ab70ccc7d3a194960c59f0075e1f98168f4477ca.

File: src/app/doctor/services/usecases_test.go (hunk 4 above)
Comment (thisham) [reply src/app/doctor/services/usecases_test.go]: applied https://github.com/ALTERA-CAPSTON-41/backend/pull/28/commits/ab70ccc7d3a194960c59f0075e1f98168f4477ca.

**Pull request merged**